src/Extensions/S3/Program.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using System.Text.Json;
+
+namespace Microsoft.Diagnostics.Monitoring.S3
+{
+    internal sealed class Program
```

Comment:
Because this is >95% a copy of what's in the Azure extension, I'd like to see this moved to the Common project as a utility method that you can parameterize (passing it the args from the Main entrypoint, specifying the egress provide via generic parameter, etc).

Reply:
Same question as above

Reply:
Same response as above